lib/iris/fileformats/grib/__init__.py

```diff
@@ -239,7 +239,7 @@ def __init__(self, grib_message, grib_fh=None, auto_regularise=True):
             # The byte offset requires to be reset back to the first byte
             # of this message. The file pointer offset is always at the end 
             # of the current message due to the grib-api reading the message.
-            proxy = GribDataProxy(shape, np.zeros(0).dtype, np.nan,
+            proxy = GribDataProxy(shape, np.array([0.]).dtype, np.nan,
```

Comment:
When I spoke to Dan about this he suggested that this is because at some point in the future, numpy might mess about with np.float as a datatype, so this method is more robust.